### PR TITLE
Issue 1891: Add role to Signal for custom zero logic

### PR DESCRIPTION
**Pull request by jstuder-gh**
This add custom logic for ```Signal(0)``` so that it will return the type object
like it did previously.

See [Issue 1891](https://github.com/rakudo/rakudo/issues/1891)

I set the target of the PR to the post-release branch, but if it makes more
sense to merge it into master before release, then I am okay with it being
changed.

### CODE REVIEW

**Comment by lizmat** [comment]
Not sure it is a good idea to make CALL-ME a multi-sub

File: src/core/signals.pm6 [hunk 1]
```diff
@@ -1,4 +1,10 @@
-my enum Signal (
+my role Signally {
+    multi method CALL-ME(Int $signum) {
```
Comment (AlexDaniel) [inline src/core/signals.pm6]: But this won't cover other reasonable cases like `Signal(‘0’)`?

Comment (jstuder-gh) [reply src/core/signals.pm6]: Thanks. I've pushed up a new commit that allows for scenarios like this.

**Comment by AlexDaniel** [comment]
Let's try that.

@jstuder-gh There's an interesting idea [here](http://colabti.org/irclogger/irclogger_log/perl6-dev?date=2018-06-17#l135), perhaps we should consider reworking it later.

**Comment by AlexDaniel** [comment]
I think there's a different issue (?) now:

> \<AlexDaniel\> c: 2018.05,HEAD say Signal(∞)
> \<committable6\> AlexDaniel, ¦2018.05: «(Signal)␤» ¦HEAD(5a6ff40): «Cannot convert Inf to Int: ␤  in block <unit> at /tmp/5jwSYHlfGp line 1␤␤ «exit code = 1»»

I don't know if it's important, but I still wonder if there's any easy way to keep previous behavior.

**Comment by jstuder-gh** [comment]
@AlexDaniel, I pushed another commit to the same branch that adds an Inf candidate, which you can merge in if you'd like.

**Pull request merged**